### DISCUSSION TITLE
Great job

### DISCUSSION THREAD

**Discussion by aven187** [sb0y/warp_beacon#8] [General]
Hey mate,

First of all great job, any way to tip you for your work ? 

Really powerful and it helps me to download instagram posts. 

Do you have any ideas how instagram would potentially block our accounts ? I know for example, when there is suspicious activity, they will challenge you with a bot/captcha to authenticate for example through a browser, it would mean break the bot. Don't know if we should use the same user agent than the browser we used to create the account.

For now, I've trying to do this to avoid to get into this situation: 

1. VPS with Ubuntu and Graphical desktop 
2. Created my account from this public IP (looking maybe to get a residential proxy instead of datacenter ip) 
3. Put profile picture, bio, following people and try to act as a normal user for the next 2 weeks by liking some posts
4. don't connect this account in any other way (my smartphone, other browser, other ip, etc)

Looking forward to see your feedback

**Comment by sb0y** [comment]
Hi, @aven187!
Donations are accepted via the [link](https://pay.cryptocloud.plus/pos/W5BMtNQt5bJFoW2E). :)

Instagram uses advanced algorithms to track suspicious activity.
The first thing you need to know is that it's recommended to use old accounts with a long history.

To reduce suspicious activity, I use a scheduler that randomly performs actions mimicking a real person.

Additionally, in case of a ban, it's recommended to use multiple accounts linked to different IP addresses.

To change the IP address, you can use a newly introduced [proxy mechanism](https://github.com/sb0y/warp_beacon/wiki/Introduction-in-account.json-file).

In case of any issues with an account, the bot will send a notification to the bot administrator and attempt to complete the task using the next account in the list. Accounts are linked to proxies.

Additionally, the maintainer of the instagrapi library, @subzeroid has announced a feature for solving captchas, which I'm eagerly awaiting.

Overall, the points you mentioned are reasonable and are common practice.
